src/main/java/com/hedera/exchange/ERTproc.java

```diff
@@ -139,10 +139,17 @@ public ExchangeRate call() {
                     (int) (medianExRate * 100 * this.hbarEquiv),
                     nextExpirationTimeInSeconds);
 
-            if(midnightExchangeRate != null && !midnightExchangeRate.isSmallChange(this.bound, nextRate)) {
-                LOGGER.debug(Exchange.EXCHANGE_FILTER, "last midnight value present. Validating the median with {}",
-                        midnightExchangeRate.toJson());
-                    nextRate = midnightExchangeRate.clipRate(nextRate, this.bound);
+            if(midnightExchangeRate != null) {
+                if(!midnightExchangeRate.getNextRate().isSmallChange(this.bound, nextRate)) {
+                    LOGGER.debug(Exchange.EXCHANGE_FILTER, "last midnight value present. Validating the nextRate with {}",
+                            midnightExchangeRate.getNextRate().toJson());
+                    nextRate = midnightExchangeRate.getNextRate().clipRate(nextRate, this.bound);
+                }
+                if(!midnightExchangeRate.getCurrentRate().isSmallChange(this.bound, currentExchangeRate)) {
+                    LOGGER.debug(Exchange.EXCHANGE_FILTER, "last midnight value present. Validating the currentRate with {}",
+                            midnightExchangeRate.getCurrentRate().toJson());
+                    currentExchangeRate = midnightExchangeRate.getCurrentRate().clipRate(currentExchangeRate, this.bound);
```

Comment:
wouldn't this one already have been clipped? By the previous run?

Reply:
yes, in the previous Run it is clipped when compared to nextRate of MidnightExchangeRate, now it is clipped by comparing it to currentRate of MidnightExchangeRate